pkg/datapath/linux/config/config_test.go

```diff
@@ -108,6 +110,88 @@ func (s *ConfigSuite) TestWriteEndpointConfig(c *C) {
 	writeConfig(c, "endpoint", func(w io.Writer, dp datapath.ConfigWriter) error {
 		return dp.WriteEndpointConfig(w, &dummyEPCfg)
 	})
+
+	testRun := func(t *testutils.TestEndpoint) ([]byte, map[string]uint32, map[string]string) {
+		cfg := &HeaderfileWriter{}
+		varSub, stringSub := loader.ELFSubstitutions(t)
+
+		var buf bytes.Buffer
+		cfg.writeStaticData(&buf, t)
+
+		return buf.Bytes(), varSub, stringSub
+	}
+
+	lxcIPs := []string{"LXC_IP_1", "LXC_IP_2", "LXC_IP_3", "LXC_IP_4"}
+
+	tests := []struct {
+		description string
+		template    testutils.TestEndpoint // Represents template bpf prog
+		endpoint    testutils.TestEndpoint // Represents normal endpoint bpf prog
+		preTestRun  func(t *testutils.TestEndpoint, e *testutils.TestEndpoint)
+		templateExp bool
+		endpointExp bool
+	}{
+		{
+			description: "IPv6 is disabled, endpoint does not have an IPv6 addr",
+			template:    testutils.NewTestEndpoint(),
+			endpoint:    testutils.NewTestEndpoint(),
+			preTestRun: func(t *testutils.TestEndpoint, e *testutils.TestEndpoint) {
+				option.Config.EnableIPv6 = false
```

Comment:
@cilium/ci folks, is this best practice or do you have an alternative suggestion here?

Reply:
I can see how that could pollute the other tests. One way could be to have a `postTestRun` cleanup function to set the option back to its original value.

Reply:
The leakage would be my primary concern. I don't know if we have too much by way of best-practice in the unit tests. I suspect that they are scoped to each package so you shouldn't be re-initialising stuff out-of-order with this setup. What don't know is what happens when someone realises that we can run unit tests in parallel.

Is it possible to simply change how the components we test are initialised so that they take the ipv6 parameter instead of wiring in the global?

Reply:
Hmm, it is possible, but my quick judgements say that it would require a major refactor.

Reply:
Sounds like restoring the option at the end of the test is a reasonable middle-ground.

We can keep thinking about the best way to represent the options throughout packages going forward with this in mind.

Reply:
Done